nexusserver/server.go

```diff
@@ -0,0 +1,463 @@
+		// Defaults to output with indentation.
+		Marshaler nexusapi.Marshaler
+		// Max duration to allow waiting for a single get result request.
+		// Enforced if the provided wait query parameter for a request is greater than this value.
+		//
+		// Defaults to one minute.
+		GetResultMaxRequestTimeout time.Duration
+	}
+
+	// Input for [nexusserver.Handler.StartOperation].
+	StartOperationRequest struct {
+		// Operation name.
+		Operation string
+		// Request ID, should be used to dedupe start requests.
+		RequestID string
+		// Callback URL to call upon completion if the started operation is async.
+		CallbackURL string
+		// The original HTTP request.
+		// Read the URL, Header, and Body of the request to process the operation input.
+		HTTPRequest *http.Request
+	}
+
+	// Input for [nexusserver.Handler.GetOperationResult].
+	GetOperationResultRequest struct {
+		// Operation name.
+		Operation string
+		// Operation ID as originally generated by a Handler.
+		// It is the handler's responsibility to validate this ID and authorize access to the underlying resource.
+		OperationID string
+		// Whether the caller has indicated that it wants to wait for operation completion, turning the request into a
+		// long poll. When set, the wait duration will be reflected in the provided context deadline.
+		Wait bool
+		// The original HTTP request.
+		HTTPRequest *http.Request
+	}
+
+	// Input for [nexusserver.Handler.GetOperationInfo].
+	GetOperationInfoRequest struct {
+		// Operation name.
+		Operation string
+		// Operation ID as originally generated by a Handler.
+		// It is the handler's responsibility to validate this ID and authorize access to the underlying resource.
+		OperationID string
+		// The original HTTP request.
+		HTTPRequest *http.Request
+	}
+
+	// Input for [nexusserver.Handler.CancelOperation].
+	CancelOperationRequest struct {
+		// Operation name.
+		Operation string
+		// Operation ID as originally generated by a Handler.
+		// It is the handler's responsibility to validate this ID and authorize access to the underlying resource.
+		OperationID string
+		// The original HTTP request.
+		HTTPRequest *http.Request
+	}
+
+	// A response to a StartOperation or GetResult request.
+	OperationResponse interface {
+		applyToStartResponse(http.ResponseWriter, *httpHandler)
+		applyToGetResultResponse(http.ResponseWriter, *httpHandler)
+	}
+
+	// Return this error from any handler method to fail a request with a custom status code and failure message.
+	HandlerError struct {
+		// Defaults to 500.
+		StatusCode int
+		// Failure to report back in the response.
+		Failure *nexusapi.Failure
+	}
+
+	// A Handler implements all of the Nexus service endpoints as defined in the Nexus HTTP API (https://github.com/nexus-rpc/api).
+	Handler interface {
+		StartOperation(context.Context, *StartOperationRequest) (OperationResponse, error)
+		GetOperationResult(context.Context, *GetOperationResultRequest) (OperationResponse, error)
```

Comment:
```suggestion
		StartOperation(context.Context, *StartOperationRequest) (StartOperationResponse, error)
		GetOperationResult(context.Context, *GetOperationResultRequest) (OperationResult, error)
```

No need to force `GetOperationResult` to return `OperationResponseSync` or something as if it was a sync operation I don't think